tower/src/builder/mod.rs

```diff
@@ -212,6 +212,11 @@ impl<L> ServiceBuilder<L> {
         self.layer(TimeoutLayer::new(timeout))
     }
 
+    /// Obtains the underlying `Layer` impementation.
+    pub fn into_stack(self) -> L {
```

Comment:
Right, but we have another method `layer()` that takes a layer, but that's not exactly what we're returning... We're generally returning a `Stack<...>` here.

@carllerche @seanmonstar wdyt is clearest here?

Reply:
Specifically, I think it would be weird to read:
```rust
ServiceBuilder::default()
  .layer(..)
  .layer(..)
  .into_layer()
```

Reply:
`into_stack` seems fine, or `into_inner`.

Reply:
`into_inner` technically works here because `ServiceBuilder<T>` is just a wrapper around `T`. It feels a bit weird in this context... but I don't have any better ideas, so I would vote to just go w/ `into_inner`.

Reply:
Some other options to consider:
* `make_layer()`
* `build_layer()`
* `flatten()` 🙃 

Could also add a constraint to the impl so the method is only available when `L: Layer<S>` if that makes naming any easier.

Reply:
> Could also add a constraint to the impl so the method is only available when L: Layer<S> if that makes naming any easier.

This would be awkward, since into_inner would have to depend on an S but doesn't take one, users would frequently be required to name the service type...